gprofiler/utils/__init__.py

```diff
@@ -125,20 +133,32 @@ def start_process(
             env = env if env is not None else os.environ.copy()
             env.update({"LD_LIBRARY_PATH": ""})
 
-    cur_preexec_fn = kwargs.pop("preexec_fn", os.setpgrp)
+    if not is_windows():
+        cur_preexec_fn = kwargs.pop("preexec_fn", os.setpgrp)
+    else:
+        cur_preexec_fn = kwargs.pop("preexec_fn", None)
 
     if term_on_parent_death:
         cur_preexec_fn = wrap_callbacks([set_child_termination_on_parent_death, cur_preexec_fn])
 
-    popen = Popen(
-        cmd,
-        stdout=kwargs.pop("stdout", subprocess.PIPE),
-        stderr=kwargs.pop("stderr", subprocess.PIPE),
-        stdin=subprocess.PIPE,
-        preexec_fn=cur_preexec_fn,
-        env=env,
-        **kwargs,
-    )
+    if is_windows():
+        popen = Popen(
+                    cmd, 
+                    stdout=kwargs.pop("stdout", subprocess.PIPE),
+                    stderr=kwargs.pop("stderr", subprocess.PIPE),
+                    stdin=subprocess.PIPE,
+                    env=env,
+                    **kwargs,)
+    else:
+        popen = Popen(
+                    cmd,
+                    stdout=kwargs.pop("stdout", subprocess.PIPE),
+                    stderr=kwargs.pop("stderr", subprocess.PIPE),
+                    stdin=subprocess.PIPE,
+                    preexec_fn=cur_preexec_fn,
```

Comment:
Is this change needed given that you only set `cur_preexec_fn` to `os.setpgrp` in the `not is_windows()` case? (I refer to this `is_windows()` branch from line 144)